### PR TITLE
API-15744/ Swagger friendly OAS

### PR DESCRIPTION
**Pull request by ediiotero**
Story: [API-18120](https://vajira.max.gov/browse/API-18120) also fixes problems with [API-15744](https://vajira.max.gov/browse/API-15744)

GET /docserver/apis/<apiName>/versions/<apiVersion>/oas/<oasIteration>
GET /docserver/apis/<apiName>/versions/<apiVersion>/oas/latest

work as expected only returns iteration info.
SF OAS is working locally

### CODE REVIEW

File: src/oas/oas.service.ts [hunk 1]
```diff
@@ -130,13 +149,22 @@ class OASService {
       StatusEnum.INACTIVE,
       StatusEnum.ERROR,
     ],
+    isFullDetailsRequired?: boolean,
   ): Promise<OASModel[]> {
     try {
+      let populate: any[] = [];
+      isFullDetailsRequired = false;
```
Comment (mheusgit) [inline src/oas/oas.service.ts:156]: Needs to be removed

File: src/oas/oas.service.ts [hunk 2]
```diff
@@ -130,13 +149,22 @@ class OASService {
       StatusEnum.INACTIVE,
       StatusEnum.ERROR,
     ],
+    isFullDetailsRequired?: boolean,
```
Comment (mheusgit) [inline src/oas/oas.service.ts:152]: A default can be defined here

File: src/oas/oas.service.ts [hunk 3]
```diff
@@ -149,20 +149,15 @@ class OASService {
...
-
-      if (isFullDetailsRequired) {
-        populate = ['spec', 'transformedSpec', 'zippedSFTransformedSpec'];
-      }
-
       await apiVersion.openAPISpecs.init({
         orderBy: { iteration: 'desc' },
         where: { status: { $in: status } },
-        populate: populate,
+        populate: isFullDetailsRequired
```
Comment (mheusgit) [inline src/oas/oas.service.ts:158]: Even better

**Review by mheusgit — COMMENTED** [review]
I'd approve this PR if I didn't write some of the changes

**Review by peyton-lucas — APPROVED** [review]
LGTM

**Comment by mheusgit** [comment]
Looks like the wrong ticket number was used.  Should have been API-18120

**Pull request merged**